### PR TITLE
contrib/patchutils: new package (0.4.2)

### PR DESCRIPTION
**Pull request by reocat**

### CODE REVIEW

File: contrib/patchutils/template.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+source = (
+    f"http://cyberelk.net/tim/data/patchutils/stable/patchutils-{pkgver}.tar.xz"
+)
+sha256 = "8875b0965fe33de62b890f6cd793be7fafe41a4e552edbf641f1fed5ebbf45ed"
+# 6 tests fail
+options = ["!check"]
+
+
+def do_configure(self):
+    self.do(self.chroot_cwd / "configure", "--prefix=/usr")
```
Comment (nekopsykose) [inline contrib/patchutils/template.py:23]: build_style = configure does this for you

File: contrib/patchutils/template.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+pkgname = "patchutils"
+pkgver = "0.4.2"
+pkgrel = 2
```
Comment (nekopsykose) [inline contrib/patchutils/template.py:3]: pkgrel starts at 0

File: contrib/patchutils/template.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
+pkgname = "patchutils"
+pkgver = "0.4.2"
+pkgrel = 0
+build_style = "configure"
```
Comment (q66) [inline contrib/patchutils/template.py:4]: is gnu_configure

File: contrib/patchutils/template.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+configure_args = ["--prefix=/usr"]
```
Comment (q66) [inline contrib/patchutils/template.py:5]: is unnecessary with gnu_configure

File: contrib/patchutils/template.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+make_cmd = "gmake"
+make_dir = "."
+hostmakedepends = ["gmake"]
+makedepends = ["xmlto"]
+depends = ["perl"]
+pkgdesc = "Collection of programs for manipulating patch files"
+maintainer = "reocat <ng.ct_ml@tuta.io>"
+license = "GPL-2.0-or-later"
+url = "https://cyberelk.net/tim/patchutils"
```
Comment (q66) [inline contrib/patchutils/template.py:14]: url doesn't work over https

File: contrib/patchutils/template.py [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+source = (
+    f"http://cyberelk.net/tim/data/patchutils/stable/patchutils-{pkgver}.tar.xz"
+)
+sha256 = "8875b0965fe33de62b890f6cd793be7fafe41a4e552edbf641f1fed5ebbf45ed"
+# 6 tests fail
```
Comment (q66) [inline contrib/patchutils/template.py:19]: tests should be patched not to fail

File: contrib/patchutils/template.py [hunk 7]
```diff
@@ -0,0 +1,20 @@
+pkgname = "patchutils"
+pkgver = "0.4.2"
+pkgrel = 0
+build_style = "configure"
+configure_args = ["--prefix=/usr"]
+make_cmd = "gmake"
+make_dir = "."
+hostmakedepends = ["gmake"]
+makedepends = ["xmlto"]
```
Comment (q66) [inline contrib/patchutils/template.py:9]: is a runnable program build-time dependency, thus hostmakedepends

Comment (q66) [reply contrib/patchutils/template.py:9]: also likely want to build with pcre2 support, thus pcre2-devel in makedepends

**Comment by q66** [comment]
a bunch of the bundled scripts also use a `#!/bin/bash` shebang despite seemingly using no bashisms

**Comment by q66** [comment]
before this is ready, these things are pending:

1) the scripts i mentioned above need correcting and testing
2) while disabling failing tests is potentially fine, i need to know why they fail, so you need to post the detailed test suite failure logs

**Comment by reocat** [comment]
>     2. while disabling failing tests is potentially fine, i need to know why they fail, so you need to post the detailed test suite failure logs (in fact i strongly suspect they fail because of missing bash for the uncorrected scripts)

It's not looking like the reason is bash, or at least not in every failed tests. I tried to add bash in dependencies and removing patch, but it still the same. So I guess there potentially could be some another reasons for that. Maybe I'm wrong. Anyways, log from test suite:

[test-suite.log](https://github.com/chimera-linux/cports/files/13401463/test-suite.log)

**Comment by q66** [comment]
yes these failures are unrelated, let's keep them patched out for now

the scripts still need fixing though

**Comment by reocat** [comment]
Screenshot from 2023-11-19 11-23-11
And this was in sources:
Screenshot from 2023-11-19 11-23-30
Perl shebang getting ignored, causing errors like this:
`zsh: exec format error: unwrapdiff`
Should i try to hardcode path to perl?

**Comment by q66** [comment]
put perl in hostmakedepends and it may work (re-check the tests with it too)

**Comment by reocat** [comment]
Okay, looks like problem with perl was from my side and not linked to package. I guess i just forgot to update package

**Pull request merged**